python/sdssdb/connection.py

```diff
@@ -60,18 +60,25 @@ class DatabaseConnection(six.with_metaclass(abc.ABCMeta)):
     autoconnect : bool
         Whether to autoconnect to the database using the profile parameters.
         Requites `.dbname` to be set.
+    dbversion : str
+        A database version.  If specified, appends to dbname as "dbname_dbversion"
+        and becomes the dbname used for connection strings.
 
     """
 
     #: The database name.
     dbname = None
+    dbversion = None
 
-    def __init__(self, dbname=None, profile=None, autoconnect=True):
+    def __init__(self, dbname=None, profile=None, autoconnect=True, dbversion=None):
 
         #: Reports whether the connection is active.
         self.connected = False
         self.profile = None
         self.dbname = dbname if dbname else self.dbname
+        self.dbversion = dbversion or self.dbversion
+        if self.dbversion:
+            self.dbname = f'{self.dbname}_{self.dbversion}' 
```

Comment:
This dbversion thing seems a bit adhoc and assumes a given format for the versions of the databases. I don't love it.

How is this different from having a default version for a database (supposedly, the latest) and if you want a different one you call `connect` with the new database name?

Reply:
It is a bit wonky I know.  I added this to deal with the `archive` database.  Each data release, a new database is created with names like `archive_20200203`, or `archive_20190711` but the models and connection don't change.  In principle there's nothing wrong with fixing the database name to the one with the latest version but I didn't like the idea of always having to commit a new change for that and potentially tag a new release.    

We don't have a strong policy yet about versioning databases and/or version naming schemes but it might be a good idea to make one.  It makes sense to me to somehow separate db names and versions. But I'm open to suggestions.